main/release/releasability/releasability.py

```diff
@@ -0,0 +1,196 @@
+        return correlation_id
+
+    def _build_sns_request(self,
+                           correlation_id: str,
+                           organization: str,
+                           project_name: str,
+                           branch_name: str,
+                           revision: str,
+                           version: str,
+                           build_number: int):
+        sns_request = {
+            'uuid': correlation_id,
+            'responseToARN': self.RESULT_TOPIC_ARN,
+            'repoSlug': f'{organization}/{project_name}',
+            'version': version,
+            'vcsRevision': revision,
+            'artifactoryBuildNumber': build_number,
+            'branchName': branch_name
+        }
+        return sns_request
+
+    """
+    This method is responsible to return an SQS queue url based on an arn.
+    More details about SQS urls can be found here:
+    https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-queue-message-identifiers.html
+    """
+
+    @staticmethod
+    def _arn_to_sqs_url(arn):
+        parts = arn.split(':')
+        service = parts[2]
+
+        if service != "sqs":
+            raise ValueError(f"Invalid sqs ARN: {arn}")
+
+        region = parts[3]
+        account_number = parts[4]
+        queue_name = parts[5]
+        return f'https://sqs.{region}.amazonaws.com/{account_number}/{queue_name}'
+
+    def get_releasability_report(self, correlation_id: str) -> ReleasabilityChecksReport:
+
+        check_results = self._get_check_results(correlation_id)
+        return ReleasabilityChecksReport(check_results)
+
+    def _get_check_results(self, correlation_id: str):
+        filters = self._build_filters(correlation_id)
+
+        expected_message_count = self._get_checks_count()
+        received_message_count = 0
+        received_check_results = list[ReleasabilityCheckResult]()
+
+        now = time.time()
+        while (received_message_count < expected_message_count
+               and not has_exceeded_timeout(now, Releasability.FETCH_CHECK_RESULT_TIMEOUT_SECONDS)):
+            filtered_messages = self._fetch_filtered_check_results(filters)
+
+            for message_payload in filtered_messages:
+                received_message_count += 1
+
+                received_check_results.append(
+                    ReleasabilityCheckResult(
+                        message_payload["checkName"],
+                        message_payload["type"],
+                        message_payload["message"] if "message" in message_payload else None
+                    )
+                )
+
+            time.sleep(Releasability.FETCH_SLEEP_TIME_SECONDS)
+
+        #  TODO: The code should not relies on amount of results but check 1:1 between check name and received response.
+        #  A ticket has been created for that: https://sonarsource.atlassian.net/browse/BUILD-4703
```

Comment:
One-line TODO is better parsed and displayed
```suggestion
        #  TODO BUILD-4703: The code should not rely on amount of results but check 1:1 between check name and received response.
```

Reply:
This violates pre-commit requirements, I will remove this todo as after evaluating this is unfortunately not doable :+1: